manifests/daemon_reload.pp

```diff
@@ -0,0 +1,42 @@
+# @summary Run systemctl daemon-reload
+#
+# @api public
+#
+# @param name
+#   A globally unique name for the resource
+#
+# @param enable
+#   Enable the reload exec
+#
+#   * Added in case users want to disable the reload globally using a resource collector
+#
+# @param lazy_reload
+#   Enable a global lazy reload.
+#
+#   * This is meant for cleaning up systems that may have gotten out of sync so no particular
+#     care is taken in trying to actually "fix" things, that would require a custom type that
+#     manipulates the running catalog tree.
+#
+define systemd::daemon_reload (
+  Boolean $enable      = true,
+  Boolean $lazy_reload = false,
+) {
+  if $enable {
+    if $lazy_reload {
+      exec { "${module_name}-${name}-global-systemctl-daemon-check":
+        command => 'systemctl daemon-reload',
+        onlyif  => 'systemctl show "*" --property=NeedDaemonReload | grep -q "=yes"',
```

Comment:
May I suggests using `systemctl show "*" --property=NeedDaemonReload | grep -qxFm1 "NeedDaemonReload=yes"`

Reply:
@simondeziel Since it's the only field being returned we don't need an exact match. If there's a `=yes` anywhere in there, it's go time.

Reply:
🤔 Definitely could stop at the first match though

Reply:
Yeah, the efficiency makes sense. Will update.

Reply:
That is definitely nitpick territory but just found out about the `--value` option: `systemctl show "*" --value --property=NeedDaemonReload | grep -qxFm1 "yes"`

Reply:
@simondeziel Let me see if it works on EL7

Reply:
`systemd` has added so many options over time I get nervous about any but the basics across multiple OS versions

Reply:
TIL, thanks for testing!